### ISSUE TITLE
Except Validation

### ISSUE THREAD

**Issue by shine2lay** [mergeability/mergeable#62]
If you want this feature, please react with a 👍

---
Context
---
Allow exception to certain rules, for example, if title contains `(fix)`, don't validate milestone check.

Solution
--- 
To be discussed in comments

**Comment by jusx** [comment]
Possible implementation is to have an `unless` block:

```yml
- do: milestone
   must_include: version 1
   unless:
      - do: title
         must_include: WIP
```

Other use cases are described here: #145

**Comment by jusx** [comment]
@shine2lay Also, wouldn't it be nicer to just write `includes:` the word `must` in `must_include` seem redundant. Just a thought :-)

**Comment by roblabla** [comment]
Currently implementing this and hitting a few roadblocks. The given syntax doesn't really work. Different options have an implicit `and` relationship, whereas `unless` would need an implicit `or` relationship. So I could do:

```yaml
- do: milestone
   unless:
      must_include: 
        regex: version 1
      validators:
      - do: title
        must_include: 
          regex: WIP
```

Which I find fairly ugly. Or I could do:

```yaml
- do: milestone
  or:
  - must_include: 
      regex: version 1
  - run:
      do: title
      must_include: 
        regex: WIP
```

I'm currently trying to implement this second one. So there will be a `run` option that simply calls [getValiatorPromises](https://github.com/mergeability/mergeable/blob/master/lib/flex.js#L122) (nice typo :') ) and awaits it, passing through the result.

I'm having a bit of an issue figuring out how to get access to the registry though. Should I pass it down the call graph? Should I recreate it within `run` ?

**Comment by shine2lay** [comment]
@roblabla Thanks for catching the typo :1st_place_medal: ! In terms of design, do you think having the 'or' and 'and' at the `do` level might be better? Here is the config i am thinking of
```
- do: or
    - do : milestone
        must_include:
          regex: version 1
     - do : title
        must_include:
        regex: WIP
```
I think this config will achieve the 2nd scenario you mentioned but also open up a lot more use cases of mix and matching different validators.

**Comment by roblabla** [comment]
I do think having an or/and at the validator level makes more sense, and it is probably easier to implement too. Turns out making options.process async is a very invasive change, and that’s been giving me trouble.

**Comment by jusx** [comment]
resolved by #227
